awx_collection/plugins/modules/tower_project_update.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_project_update
+author: "Sean Sullivan (@sean-m-sullivan)"
+short_description: Sync a Project in Ansible Tower
+description:
+    - Sync an Ansible Tower Project. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - The name of the project to update.
+      required: True
+      type: str
+      aliases:
+        - project
+    organization:
+      description:
+        - Organization the project exists in.
+        - Used to help lookup the object, cannot be modified using this module.
+        - If not provided, will lookup by name only, which does not work with duplicates.
+      type: str
+    wait:
+      description:
+        - Wait for the project to update.
+      default: True
+      type: bool
+    interval:
+      description:
+        - The interval to request an update from Tower.
+      required: False
+      default: 1
+      type: float
+    timeout:
+      description:
+        - If waiting for the project to update this will abort after this
+          amount of seconds
+      type: int
+extends_documentation_fragment: awx.awx.auth
+'''
+
+RETURN = '''
+project_info:
+    description: dictionary containing information about the project updated
+    returned: If project synced
+    type: dict
```

Comment:
I'm torn on this return. We do a similar dict return in tower_workflow_launch. However, in tower_job_cancel, tower_job_launch, and tower_job_wait we are returning specific items (i.e. id, time elapsed, status, etc). I think I like returning specific documented things better than a generic hash with no definition of what is in it. Thoughts?

Reply:
It looks like below we are returning specific pieces of information, not a project_info dict.